### ISSUE TITLE
additional variables?

### ISSUE THREAD

**Issue by johnsonice** [jasonwengwei/cryptocurrency-prediction#2]
Nice try. it is a difficult, and most people will say it is impossible. 

A couple of things you may want to think about. 

right now, your forecast horizon is basically 1 day. You may want to make your model be able to predict multiple forecast horizon. say 1 day to a week. that will make it more useful, although the further you extend your horizon, the worse prediction you will get. 

A bigger issue here is your testing loss (0.01479) is way bigger then you eval loss (0.0021). This basically mean your model is over fitting. so you only have 800+ observations, but you have 272,931 trainable params, so there are so many freedom to let the model easily fit your small training data. So it is difficult to convince people that you model actually understands the pattern of how the price moves, it is basically overfitting it. Maybe you want to slim down your model or higher dropout etc. 

Thirdly, you may want to find other time series that are correlated to bitcoin price. It is actually very difficult to simply use past price to predict future price. There is basically no theory supporting this. You may want to find some other variables, so called "leading indicators". it is very common in macro economic forecasting. Of course it is difficult to find, if it is easy, everybody can predict. 

As right now, most of the price up and downs are driven by speculators, I am thinking if it is possible to extract some daily investors' sentiment towards bitcoin. If you can form some thing like that, you will have a very convincing story. Investors' sentiment will have a direct correlation with bitcoin demand, as bitcoin supply is basically fixed. So higher demand will basically lead to higher price and vice versa. that is the only thing I can think of right now, but you can definitely explore more. For instance, bitcoin price is very sensitive to regulations, for instance Chinese government announces that it will ban bitcoin transition, the price fell immediately. You can think if it is possible to incorporate those information in your model as well. 

in general, it is very very hard. I don't even know if it is possible at all. Things like regulation change are basically impossible to predict. But it should be a good practice. 

Have fun and let me know if i can help with anything.
